pkg/search/postgres/common.go

```diff
@@ -110,30 +110,32 @@ func (q *query) ExtraSelectedFieldPaths() []pgsearch.SelectQueryField {
 		return nil
 	}
 
+	seenSelectPathsToIndex := make(map[string]int)
+	for idx, f := range q.SelectedFields {
+		seenSelectPathsToIndex[f.SelectPath] = idx
+	}
+
 	var out []pgsearch.SelectQueryField
 	for _, groupBy := range q.GroupBys {
-		var alreadyExists bool
-		for idx := range q.SelectedFields {
-			field := &q.SelectedFields[idx]
-			if field.SelectPath == groupBy.Field.SelectPath {
-				field.FromGroupBy = true
-				alreadyExists = true
-			}
-		}
-		if !alreadyExists {
+		idx, found := seenSelectPathsToIndex[groupBy.Field.SelectPath]
+		if !found {
 			out = append(out, groupBy.Field)
+		} else {
+			field := &q.SelectedFields[idx]
+			field.FromGroupBy = true
 		}
+
+		// -1 because the was added to `out` and not to `q.SelectedFields`.
```

Comment:
start of comment is missing a subject `because the was`.  Same for the one below